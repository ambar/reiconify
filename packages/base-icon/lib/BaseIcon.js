import React, {forwardRef} from "react";
const ZWSP = "\u200B";
const InlineCenter = ({children, disabled = false}) => disabled ? children : /* @__PURE__ */ React.createElement("span", {
  style: {display: "inline-flex", alignItems: "center"}
}, ZWSP, children);
const BaseIcon = forwardRef(({
  size,
  as: As = "svg",
  text = false,
  center = false,
  fill = "currentColor",
  ...rest
}, ref) => {
  if (text) {
    size = "1.2em";
  }
  return /* @__PURE__ */ React.createElement(InlineCenter, {
    disabled: !center
  }, /* @__PURE__ */ React.createElement(As, {
    ref,
    ...rest,
    ...size && {width: size, height: size},
    fill
  }));
});
export default BaseIcon;
