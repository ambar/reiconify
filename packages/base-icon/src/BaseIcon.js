import React from 'react'

// use zero-width space to mock [strut](https://www.w3.org/TR/CSS22/visudet.html#strut), &#8203;
const ZWSP = '\u200b'

const BaseIcon = props => {
  let {name, className, defaultClassName, size, text, center, ...rest} = props

  if (text) {
    size = '1.2em'
  }

  const svg = (
    <svg
      className={`${defaultClassName} ${defaultClassName}--${name}${
        className ? ` ${className}` : ''
      }`}
      {...rest}
      {...size && {width: size, height: size}}
    />
  )

  return center ? (
    <span style={BaseIcon.strutStyle}>
      {ZWSP}
      {svg}
    </span>
  ) : (
    svg
  )
}

BaseIcon.defaultProps = {
  defaultClassName: 'Icon',
  fill: 'currentColor',
}

BaseIcon.strutStyle = {
  display: 'inline-flex',
  alignItems: 'center',
}

export default BaseIcon
