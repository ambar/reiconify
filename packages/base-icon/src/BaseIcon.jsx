import React, {forwardRef} from 'react'

const ZWSP = '\u200b' // use zero-width space to mock [strut](https://www.w3.org/TR/CSS22/visudet.html#strut), &#8203;

const InlineCenter = ({children, disabled = false}) =>
  disabled ? (
    children
  ) : (
    <span style={{display: 'inline-flex', alignItems: 'center'}}>
      {ZWSP}
      {children}
    </span>
  )

const BaseIcon = forwardRef(
  (
    {
      size,
      as: As = 'svg',
      text = false,
      center = false,
      fill = 'currentColor',
      ...rest
    },
    ref
  ) => {
    if (text) {
      size = '1.2em'
    }

    return (
      <InlineCenter disabled={!center}>
        <As
          ref={ref}
          {...rest}
          {...(size && {width: size, height: size})}
          fill={fill}
        />
      </InlineCenter>
    )
  }
)

export default BaseIcon
