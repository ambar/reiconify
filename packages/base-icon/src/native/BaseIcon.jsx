import React, {forwardRef} from 'react'
import * as svg from 'react-native-svg'

const BaseIcon = forwardRef(({size, ...rest}, ref) => {
  return (
    <svg.Svg ref={ref} {...rest} {...(size && {width: size, height: size})} />
  )
})

export default BaseIcon
