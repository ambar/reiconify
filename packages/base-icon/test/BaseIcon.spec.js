import React from 'react'
import renderer from 'react-test-renderer'
import BaseIcon from '../src/BaseIcon'

describe('BaseIcon', () => {
  it('renders icon', () => {
    const tree = renderer
      .create(
        <>
          <BaseIcon>
            <path />
          </BaseIcon>
          <BaseIcon className="MyIcon">
            <path />
          </BaseIcon>
          <BaseIcon fill={'red'}>
            <path />
          </BaseIcon>
          <BaseIcon size={40}>
            <path />
          </BaseIcon>
          <BaseIcon width={10} height={20}>
            <path />
          </BaseIcon>
          <BaseIcon text center>
            <path />
          </BaseIcon>
          <BaseIcon text>
            <path />
          </BaseIcon>
          <BaseIcon center>
            <path />
          </BaseIcon>
        </>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
