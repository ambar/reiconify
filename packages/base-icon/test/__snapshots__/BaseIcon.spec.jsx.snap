// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseIcon renders icon 1`] = `
Array [
  <svg
    fill="currentColor"
  >
    <path />
  </svg>,
  <svg
    className="MyIcon"
    fill="currentColor"
  >
    <path />
  </svg>,
  <svg
    fill="red"
  >
    <path />
  </svg>,
  <svg
    fill="currentColor"
    height={40}
    width={40}
  >
    <path />
  </svg>,
  <svg
    fill="currentColor"
    height={20}
    width={10}
  >
    <path />
  </svg>,
  <span
    style={
      Object {
        "alignItems": "center",
        "display": "inline-flex",
      }
    }
  >
    ​
    <svg
      fill="currentColor"
      height="1.2em"
      width="1.2em"
    >
      <path />
    </svg>
  </span>,
  <svg
    fill="currentColor"
    height="1.2em"
    width="1.2em"
  >
    <path />
  </svg>,
  <span
    style={
      Object {
        "alignItems": "center",
        "display": "inline-flex",
      }
    }
  >
    ​
    <svg
      fill="currentColor"
    >
      <path />
    </svg>
  </span>,
]
`;
